File: internal/check/check_test.go
package check

import (
	"testing"

	"github.com/becomeless/cc-x/internal/config"
	"github.com/becomeless/cc-x/internal/i18n"
)

// classifyHTTP：状态码 -> 结果分层（OK 标志 + 文案 key）。
func TestClassifyHTTP(t *testing.T) {
	i18n.SetLang(config.LangEN)
	cases := []struct {
		code    string
		wantOK  bool
		wantKey string
	}{
		{"200", true, "check.ok"},
		{"201", true, "check.ok"},
		{"401", false, "check.auth"},
		{"403", false, "check.auth"},
		{"404", false, "check.notFound"},
		{"429", false, "check.http"},
		{"500", false, "check.http"},
	}
	for _, c := range cases {
		r := classifyHTTP(c.code)
		if r.OK != c.wantOK {
			t.Errorf("code %s: OK=%v want %v", c.code, r.OK, c.wantOK)
		}
		if want := i18n.T(c.wantKey, c.code); r.Message != want {
			t.Errorf("code %s: message=%q want %q", c.code, r.Message, want)
		}
	}
}

// authHeader：API_KEY 优先于 AUTH_TOKEN；都缺返回空。
func TestAuthHeader(t *testing.T) {
	cases := []struct {
		name string
		m    map[string]string
		want string
	}{
		{"apiKey", map[string]string{"ANTHROPIC_API_KEY": "k"}, "x-api-key: k"},
		{"token", map[string]string{"ANTHROPIC_AUTH_TOKEN": "t"}, "Authorization: Bearer t"},
		{"none", map[string]string{}, ""},
		{"apiKeyWins", map[string]string{"ANTHROPIC_API_KEY": "k", "ANTHROPIC_AUTH_TOKEN": "t"}, "x-api-key: k"},
		{"blankIgnored", map[string]string{"ANTHROPIC_API_KEY": "  ", "ANTHROPIC_AUTH_TOKEN": "t"}, "Authorization: Bearer t"},
	}
	for _, c := range cases {
		if got := authHeader(c.m); got != c.want {
			t.Errorf("%s: got %q want %q", c.name, got, c.want)
		}
	}
}

// Profile 的无网络早返回：缺地址 / 缺密钥（不触发 curl）。
func TestProfileEarlyReturns(t *testing.T) {
	i18n.SetLang(config.LangEN)

	noURL := Profile(config.Provider{Name: "X", Env: map[string]string{}})
	if noURL.OK || noURL.Message != i18n.T("check.noUrl") {
		t.Errorf("noUrl: got %+v", noURL)
	}

	noKey := Profile(config.Provider{Name: "X", Env: map[string]string{"ANTHROPIC_BASE_URL": "https://x.example.com"}})
	if noKey.OK || noKey.Message != i18n.T("check.noKey") {
		t.Errorf("noKey: got %+v", noKey)
	}
}

File: src/check.test.ts
import assert from 'node:assert/strict';
import { test } from 'node:test';

import { authHeaders, checkProfile, classifyHttp } from './check.js';
import { setLang, T } from './i18n/index.js';

// classifyHttp：状态码 -> 结果分层（ok 标志 + 文案 key）。
test('classifyHttp maps status to layered result', () => {
  setLang('en');
  const cases: Array<[number, boolean, string]> = [
    [200, true, 'check.ok'],
    [204, true, 'check.ok'],
    [401, false, 'check.auth'],
    [403, false, 'check.auth'],
    [404, false, 'check.notFound'],
    [429, false, 'check.http'],
    [500, false, 'check.http'],
  ];
  for (const [status, ok, key] of cases) {
    const r = classifyHttp(status);
    assert.equal(r.ok, ok, `status ${status} ok`);
    assert.equal(r.message, T(key, String(status)), `status ${status} message`);
  }
});

// authHeaders：API_KEY 优先于 AUTH_TOKEN；空白忽略；都缺返回 undefined。
test('authHeaders prefers API key and ignores blanks', () => {
  assert.deepEqual(authHeaders({ ANTHROPIC_API_KEY: 'k' }), { 'x-api-key': 'k' });
  assert.deepEqual(authHeaders({ ANTHROPIC_AUTH_TOKEN: 't' }), { Authorization: 'Bearer t' });
  assert.equal(authHeaders({}), undefined);
  assert.deepEqual(authHeaders({ ANTHROPIC_API_KEY: 'k', ANTHROPIC_AUTH_TOKEN: 't' }), { 'x-api-key': 'k' });
  assert.deepEqual(authHeaders({ ANTHROPIC_API_KEY: '  ', ANTHROPIC_AUTH_TOKEN: 't' }), { Authorization: 'Bearer t' });
});

// checkProfile 的无网络早返回：缺地址 / 缺密钥。
test('checkProfile early-returns without network', async () => {
  setLang('en');

  const noUrl = await checkProfile({ name: 'X', env: {} });
  assert.equal(noUrl.ok, false);
  assert.equal(noUrl.message, T('check.noUrl'));

  const noKey = await checkProfile({ name: 'X', env: { ANTHROPIC_BASE_URL: 'https://x.example.com' } });
  assert.equal(noKey.ok, false);
  assert.equal(noKey.message, T('check.noKey'));
});
